src/homework/hw18_graph/SimpleBinaryTree.java

```diff
@@ -0,0 +1,51 @@
+package homework.hw18_graph;
+
+public class SimpleBinaryTree<T extends Number> {
+
+    private NodeTree<T> root;
+    private double sum;
+
+    public SimpleBinaryTree() {
+        this.root = null;
+    }
+
+    public NodeTree<T> getRoot() {
+        return this.root;
+    }
+
+    public void add(T value) {
+        root = recursionSearchingChild(value, root);
+    }
+
+    private NodeTree<T> recursionSearchingChild(T value, NodeTree<T> node) {
+
+        if (node == null) {
+            return new NodeTree<>(value);
+        }
+
+        if (value.doubleValue() < node.getValue().doubleValue()) {
+            node.setLeftNode(recursionSearchingChild(value, node.getLeftNode()));
+        } else if (value.doubleValue() > node.getValue().doubleValue()) {
+            node.setRightNode(recursionSearchingChild(value, node.getRightNode()));
+        }
+        return node;
+    }
+
+    public double iterate(NodeTree<T> root) {
```

Comment:
If the root of the tree (root) is null, the iterate method will throw a NullPointerException.